service/controller/v5/resource/instance/resource.go

```diff
@@ -106,3 +116,34 @@ func (r *Resource) getVMsClient(ctx context.Context) (*compute.VirtualMachineSca
 
 	return sc.AzureClientSet.VirtualMachineScaleSetVMsClient, nil
 }
+
+func (r *Resource) getEncrypterObject(ctx context.Context, secretName string) (enc.Encrypter, error) {
+	r.logger.LogCtx(ctx, "level", "debug", "message", "retrieving encryptionkey")
+
+	secret, err := r.k8sClient.CoreV1().Secrets(key.CertificateEncryptionNamespace).Get(secretName, metav1.GetOptions{})
+	if err != nil {
+		return enc.Encrypter{}, microerror.Mask(err)
```

Comment:
`Encrypter` should be an interface as we want to have it replaceable. Then we can return `nil` here as zero value.